jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -2426,49 +2426,79 @@ ecma_string_substr (const ecma_string_t *string_p, /**< pointer to an ecma strin
  * Helper function for trimming.
  *
  * Used by:
- *        - ecma_string_trim
- *        - ecma_utf8_string_to_number
- *        - ecma_builtin_global_object_parse_int
- *        - ecma_builtin_global_object_parse_float
+ *        - ecma_string_trim_helper
+ *
+ * @return position of the first non whitespace character.
  */
-void
-ecma_string_trim_helper (const lit_utf8_byte_t **utf8_str_p, /**< [in, out] current string position */
-                         lit_utf8_size_t *utf8_str_size) /**< [in, out] size of the given string */
+const lit_utf8_byte_t *
+ecma_string_trim_front (const lit_utf8_byte_t *start_p, /**< current string's start position */
+                        const lit_utf8_byte_t *end_p)  /**< current string's end position */
 {
   ecma_char_t ch;
-  lit_utf8_size_t read_size;
-  const lit_utf8_byte_t *nonws_start_p = *utf8_str_p + *utf8_str_size;
-  const lit_utf8_byte_t *current_p = *utf8_str_p;
 
-  while (current_p < nonws_start_p)
+  while (start_p < end_p)
   {
-    read_size = lit_read_code_unit_from_utf8 (current_p, &ch);
+    lit_utf8_size_t read_size = lit_read_code_unit_from_utf8 (start_p, &ch);
 
     if (!lit_char_is_white_space (ch))
     {
-      nonws_start_p = current_p;
       break;
     }
-
-    current_p += read_size;
+    start_p += read_size;
   }
 
-  current_p = *utf8_str_p + *utf8_str_size;
+  return start_p;
+} /* ecma_string_trim_front */
 
-  while (current_p > nonws_start_p)
+/**
+ * Helper function for trimming.
+ *
+ * Used by:
+ *        - ecma_string_trim_helper
+ *
+ * @return position of the last non whitespace character.
+ */
+const lit_utf8_byte_t *
+ecma_string_trim_back (const lit_utf8_byte_t *start_p, /**< current string's start position */
+                       const lit_utf8_byte_t *end_p)  /**< current string's end position */
+{
+  ecma_char_t ch;
+
+  while (end_p > start_p)
   {
-    read_size = lit_read_prev_code_unit_from_utf8 (current_p, &ch);
+    lit_utf8_size_t read_size = lit_read_prev_code_unit_from_utf8 (end_p, &ch);
 
     if (!lit_char_is_white_space (ch))
     {
       break;
     }
-
-    current_p -= read_size;
+    end_p -= read_size;
   }
 
-  *utf8_str_p = nonws_start_p;
-  *utf8_str_size = (lit_utf8_size_t) (current_p - nonws_start_p);
+  return end_p;
+} /* ecma_string_trim_back */
+
+/**
+ * Helper function for trimming.
+ *
+ * Used by:
+ *        - ecma_string_trim
+ *        - ecma_utf8_string_to_number
+ *        - ecma_builtin_global_object_parse_int
+ *        - ecma_builtin_global_object_parse_float
```

Comment:
parseInt and parseFloat should only use trimStart by spec

Reply:
This is a nice catch i did not look into. Although roughly looking at them this should not alter as the code checks for digits when iterating and in the end, the `ecma_utf8_string_to_number` call the whole trim in itself as well. I'm looking into our test suites to check for fails if i modify this.